go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -253,6 +265,101 @@ func (call *builtinASCII) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Int64, Col: collationNumeric, Flag: nullableFlags(str.Flag)}, nil
 }
 
+func reverse(in *evalBytes) []byte {
+	cs := colldata.Lookup(in.col.Collation).Charset()
+	out, cur := make([]byte, len(in.bytes)), len(in.bytes)
+
+	for i := 0; i < charset.Length(cs, in.bytes); i++ {
```

Comment:
Can this loop be simplified using `DecodeRune` and `EncodeRune` on the charset instead? 

Also the loop here keeps evaluating `i < charset.Length(cs, in.bytes)` this way which is very inefficient (as this is expensive). But I think that can be avoided with using `DecodeRune` & `EncodeRune`?

Reply:
done.